examples/default/domain/aggregate.go

```diff
@@ -75,7 +75,7 @@ func (i *InvitationAggregate) HandleCommand(command eh.Command) error {
 			return nil
 		}
 
-		i.StoreEvent(&InviteAccepted{i.AggregateID()})
+		i.StoreEvent(&InviteAccepted{i.AggregateID().(eh.UUID)})
```

Comment:
The test branch looks really nice! I would love to have that solution merged if everything works with it!

I would maybe consider changing RegisterID to SetIDType or similar, just to avoid confusion that it can be done twice. Maybe an extra check so it panics if trying to set it a second time. It would also make more sense to call it in init() as a good example for how to use it.

Nice work!